File: app/src/main/java/com/example/administrator/mytemplateapplication/MainActivity.java
package com.example.administrator.mytemplateapplication;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.example.administrator.mytemplateapplication.base.BaseActivity;
import com.example.administrator.mytemplateapplication.module.test.contract.TestContract;
import com.example.administrator.mytemplateapplication.module.user.views.LoginActivity;

public class MainActivity extends BaseActivity<TestContract.Presenter> {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.Button).setOnClickListener(view -> startActivity(new Intent(getApplicationContext(), LoginActivity.class)));
        int newVis= View.SYSTEM_UI_FLAG_FULLSCREEN|View.SYSTEM_UI_FLAG_HIDE_NAVIGATION|View.SYSTEM_UI_FLAG_LAYOUT_STABLE;
        getWindow().getDecorView().setSystemUiVisibility(newVis);
    }
}

File: ktapp/src/main/java/com/example/mytemplateapplication/MainActivity.kt
package com.example.mytemplateapplication

import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import com.example.mytemplateapplication.service.DakaService
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val JOBID = 100
    lateinit var scheduler: JobScheduler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        scheduler = getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
        btStart.setOnClickListener { scheduleJob() }
        btStop.setOnClickListener {
            scheduler.cancel(JOBID)
        }
        btLogin.setOnClickListener{startActivity(Intent(this,LoginActivity::class.java))}
    }

    private fun scheduleJob() {
        val builder = JobInfo.Builder(JOBID, ComponentName(this, DakaService::class.java))
//        builder.setMinimumLatency(0)
//        builder.setOverrideDeadline(0)
        builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED)
//        builder.setRequiresDeviceIdle(false)
//        builder.setRequiresCharging(false)
        scheduler.schedule(builder.build())
    }

    override fun onTouchEvent(event: MotionEvent?): Boolean {
        Log.d("dd", event?.x.toString() + "," + event?.y)
        return super.onTouchEvent(event)
    }

}
